Feature/decay numbers

Peleccotur opened the pull request:
We had a small feature request to be able to see actual Decay values instead of percentage. Sounded like a nice place for me to start looking at code and making a small update.
Only shows in tooltip when hovering the table and on filtered DKP History. Unfiltered still shows percentage.

Peleccotur committed:
Update UserHistory tooltip to show Decay value instead of percentage

Peleccotur committed:
Update DKPHistory table to show Decay value instead of percentage when filtered on player

Taidtuskecyh commented on Modules/DKPHistory.lua:
```diff
@@ -438,9 +438,15 @@ function CommDKP:DKPHistory_Update(reset)
+					CommDKP.ConfigTab6.history[i].d:SetText("|cffff0000"..decay[#playerIndex].." "..L["DKP"].."|r - |cff616ccf"..reason.."|r |cff555555("..timeofday..")|r by |c"..c.hex..curOfficer.."|r");
+				elseif strfind(reason, L["WEEKLYDECAY"]) or strfind(reason, "Migration Correction") then
+					local decay = {strsplit(",", dkp)}
+					CommDKP.ConfigTab6.history[i].d:SetText("|cffff0000"..decay[#decay].." "..L["DKP"].."|r - |cff616ccf"..reason.."|r |cff555555("..timeofday..")|r by |c"..c.hex..curOfficer.."|r");..decay[#decay].." "..L["DKP"].."|r - |cff616ccf"..reason.."|r |cff555555("..timeofday..")|r by |c"..c.hex..curOfficer.."|r");
```
You have 2 semicolons here closing the Settext too early

Taidtuskecyh commented on TableFunctions.lua:
```diff
@@ -120,7 +120,7 @@ local function DisplayUserHistory(self, player)
 		for i=1, #PlayerSearch do
 			if not CommDKP:GetTable(CommDKP_DKPHistory, true)[PlayerSearch[i][1]].deletes and not CommDKP:GetTable(CommDKP_DKPHistory, true)[PlayerSearch[i][1]].deletedby and not CommDKP:GetTable(CommDKP_DKPHistory, true)[PlayerSearch[i][1]].hidden then
-				tinsert(PlayerTable, {reason = CommDKP:GetTable(CommDKP_DKPHistory, true)[PlayerSearch[i][1]].reason, date = CommDKP:GetTable(CommDKP_DKPHistory, true)[PlayerSearch[i][1]].date, dkp = CommDKP:GetTable(CommDKP_DKPHistory, true)[PlayerSearch[i][1]].dkp})
+				tinsert(PlayerTable, {reason = CommDKP:GetTable(CommDKP_DKPHistory, true)[PlayerSearch[i][1]].reason, date = CommDKP:GetTable(CommDKP_DKPHistory, true)[PlayerSearch[i][1]].date, dkp = CommDKP:GetTable(CommDKP_DKPHistory, true)[PlayerSearch[i][1]].dkp, players = CommDKP:GetTable(CommDKP_DKPHistory, true)[PlayerSearch[i][1]].players}})
```
One too many } at the very end :)

Commit:
fixed the semicolon and } error from previous commit

Additionally, added a optional parameter limit to TableStrFind and Table_Search functions to prevent going over few hunder / thousand records when we need last few dozen or even smaller
